### PR TITLE
added MapEditorSystem

### PR DESCRIPTION
**Pull request by andy42**

### CODE REVIEW

File: src/main/java/co/uk/genesisengineers/kitchenSink/activites/mapEditor/MapEditorActivity.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    public void onViewCreated() {
+        super.onViewCreated();
+        this.setDragbar(R.id.dragBar);
+
+        this.backButton = findViewById(R.id.backButton);
+        this.navbarTitle = (TextView)findViewById(R.id.navbarTitle);
+
+        fragmentManager.replace( R.id.layerContainer, new MapLayerFragment(mapEntity),  true);
+        Logger.info("test");
+        //fragmentManager.replace( R.id.tilePickerContainer, new TilePickerFragment(),  true);
```
Comment (differentCharacter) [inline src/main/java/co/uk/genesisengineers/kitchenSink/activites/mapEditor/MapEditorActivity.java:42]: how come this is commented out?

File: src/main/java/co/uk/genesisengineers/kitchenSink/activites/mapEditor/tileSetPickerFragment/TileSetPickerFragment.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import ui.activity.Activity;
+import ui.activity.Fragment;
+import ui.view.recyclerLayoutManager.LinearLayoutManager;
+import ui.view.RecyclerView;
+import ui.view.View;
+import ui.view.ViewGroup;
+
+import java.util.List;
+
+public class TileSetPickerFragment extends Fragment implements TileSetPickerView, TileSetPickerAdapter.OnItemSelectedListener{
```
Comment (differentCharacter) [inline src/main/java/co/uk/genesisengineers/kitchenSink/activites/mapEditor/tileSetPickerFragment/TileSetPickerFragment.java:16]: Might be too late/too much effort to change but I think "tileset" is one word.

File: src/main/java/co/uk/genesisengineers/kitchenSink/activites/recyclerViewTest/RecyclerViewActivity.java [hunk 3]
```diff
@@ -4,6 +4,8 @@
 import ui.LayoutInflater;
 import ui.activity.Activity;
 import ui.view.*;
+import ui.view.recyclerLayoutManager.GridLayoutManager;
+import ui.view.recyclerLayoutManager.LinearLayoutManager;
```
Comment (differentCharacter) [inline src/main/java/co/uk/genesisengineers/kitchenSink/activites/recyclerViewTest/RecyclerViewActivity.java:8]: Does this make `import ui.view.*;` redundant? (I'm all for importing only the specific resources needed, seems like good practice.)

File: src/main/res/drawables/icon_hide.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "type":"drawableTexture",
+  "shape":"@shapes/square_json",
+  "texture":"@textures/icon_hide_png"
+}
```
Comment (differentCharacter) [inline src/main/res/drawables/icon_hide.json:5]: there's a github warning, no newline at end of file. looks the same for the other json files too, though warnings only showing up here for some reason. I'm not sure if this actually matters in this case.

File: src/main/res/layouts/fragment_map_layer.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<LinearLayout xmlns:ge="http://www.w3.org/2001/XMLSchema-instance"
+            ge:noNamespaceSchemaLocation="layout_schema.xsd"
+            layout_width="MATCH_PARENT"
+            layout_height="WRAP_CONTENT"
+            orientation="vertical"
+            background="@color/white">
+    <TextView layout_width="MATCH_PARENT" layout_height="WRAP_CONTENT" text="Map Layers" background="@color/light_grey"/>
+    <RecyclerView id="layerRecyclerView" layout_width="MATCH_PARENT" layout_height="100"/>
+</LinearLayout>
```
Comment (differentCharacter) [inline src/main/res/layouts/fragment_map_layer.xml:9]: also newline warning on xml files - again not sure if relevant.

File: src/main/java/co/uk/genesisengineers/kitchenSink/activites/mapEditor/tileSetPickerFragment/TileSetPickerFragment.java (hunk 2 above)
Comment (andy42) [reply src/main/java/co/uk/genesisengineers/kitchenSink/activites/mapEditor/tileSetPickerFragment/TileSetPickerFragment.java:16]: i have changed it in next pull request, looks nice as well TilesetPickerFragment easier to see the difference between TilePickerFragment

File: src/main/res/drawables/icon_hide.json (hunk 4 above)
Comment (andy42) [reply src/main/res/drawables/icon_hide.json:5]: no idea why it's complaining about that when all the other files have no line brake :s

File: src/main/res/layouts/fragment_map_layer.xml (hunk 5 above)
Comment (andy42) [reply src/main/res/layouts/fragment_map_layer.xml:9]: see above

Comment (differentCharacter) [reply src/main/res/layouts/fragment_map_layer.xml:9]: The warning seems to have gone now

**Pull request merged**